dev/Dockerfile

```diff
@@ -0,0 +1,45 @@
+FROM ubuntu:18.04
+
+RUN apt-get update && \
+    apt-get install -y \
+    build-essential \
+    curl \
+    nano \
+    unzip
+
+ARG BAZEL_VERSION=0.20.0
+ARG BAZEL_OS=linux
+
+RUN curl -sL https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-installer-${BAZEL_OS}-x86_64.sh -o bazel-install.sh && \
+    bash -x bazel-install.sh && \
+    rm bazel-install.sh
+
+ARG CONDA_OS=Linux
+
+# Miniconda - Python 3.6, 64-bit, x86, latest
+RUN curl -sL https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o mconda-install.sh && \
+    bash -x mconda-install.sh -b -p miniconda && \
+    rm mconda-install.sh
+
+ENV PATH="/miniconda/bin:$PATH"
+
+ARG CONDA_ADD_PACKAGES=""
+
+RUN conda create -y -q -n tfio-dev python=3.6 ${CONDA_ADD_PACKAGES}
+
+ARG ARROW_VERSION=0.11.1
+
+RUN echo ". /miniconda/etc/profile.d/conda.sh" >> ~/.bashrc && \
+    echo "source activate tfio-dev" >> ~/.bashrc
+
+ARG PIP_ADD_PACKAGES=""
+
+RUN /bin/bash -c "source activate tfio-dev && python -m pip install \
+    pytest \
+    boto3 \
+    pyarrow==${ARROW_VERSION} \
+    pandas \
+    ${PIP_ADD_PACKAGES} \
```

Comment:
We should probably put all of the dependencies with versions in a single `requirements.txt` file. Currently they also appear in Travis scripts.

Reply:
So pyarrow is only an optional dependency to enable some functionality and the version doesn't need to be the same as Arrow C++ as long there remains binary compatibility in the format, which hasn't changed since 0.8 I believe.

Are there other hard requirements from other datasets?

Reply:
The `tensorflow-io` package itself only needs `tensorflow` package.

To test `tensorflow-io` (in `tests` directory) `pytest`, `pyarrow`, `boto3` , `pandas` (in python) and `ffmpeg` in Ubuntu are needed for now I think. Might be some others in the future.

Reply:
Should `apt-get install ffmpeg` be added to the list of packages here?

Reply:
Yes that should be enough.